src/BaseExtension.php

```diff
@@ -450,7 +450,7 @@ public function addSnippet($location, $callback, $extraparameters = [])
             $callback = (string) $callback;
         }
 
-        $this->app['asset.queue.snippet']->add($location, $callback, $this->getName(), (array) $extraparameters);
+        $this->app['asset.queue.snippet']->add($location, [$this, $callback], $this->getName(), (array) $extraparameters);
     }
```

Comment:
Ah yeah, that's right. That is why I started using `$this->app['twig']->addExtension()`